src/python/tests/genbank/test_download.py

```diff
@@ -0,0 +1,92 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Unit testing of `ensembl.io.genomio.genbank.download` module.
+
+The unit testing is divided into one test class per submodule/class found in this module, and one test method
+per public function/class method.
+
+Typical usage example::
+    $ pytest test_download.py
+
+"""
+
+from pathlib import Path
+from typing import Any
+
+import pytest
+from pytest import raises
+from unittest.mock import patch, MagicMock
+
+from ensembl.io.genomio.genbank.download import download_genbank, DownloadError
+
+
+@pytest.mark.parametrize(
+    "accession",
+    [
+        ("CM023531.1"),
+    ], 
+)
+@patch('ensembl.io.genomio.genbank.download.requests.get')
+class TestDownloadGenbank:
+    """Tests for the `download_genbank` class"""
+    
+    def test_successful_download(self, mock_requests_get: Any, tmp_dir: Path, accession: str) -> None:
+        """Tests the successful download of `download_genbank()` method.
+
+        Args:
+            mock_requests_get: A MagicMock object that mocks the request in @patch 
+            tmp_dir: Session-scoped temporary directory fixture.
+            accession: Genbank accession to be downloaded.
+        """
+
+        # Set success_code and content as an attribute to the mock object
+        mock_requests_get.return_value.status_code = 200
+        mock_content = b"The genbank download for the following accession"
+        mock_requests_get.return_value.content = mock_content
+        
+        # Temporary location where we want to store the mock output file
+        output_file = tmp_dir / f"{accession}.gb"
+        download_genbank(accession, output_file)
+
+        # Checking if the url has been called
+        mock_requests_get.assert_called_once_with(
+            "https://eutils.ncbi.nlm.nih.gov/entrez/eutils/efetch.fcgi",
+            params={"db": "nuccore", "rettype": "gbwithparts", "retmode": "text", "id": accession},
+            timeout=60
+        )
+        
+        # Assert that the content was written to the temporary file
+        with open(output_file, 'rb') as f:
+            file_content = f.read()
+        assert file_content == mock_content
+
+    def test_failed_download(self, mock_requests_failed: Any, tmp_dir: Path, accession: str) -> None:
+        """Tests the failure in downloading the files.
+
+        Args:
+            mock_requests_failed: A MagicMock object that mocks the request in @patch 
+            tmp_dir: Session-scoped temporary directory fixture.
+            accession: Genbank accession to be downloaded.
+        """
+
+        output_file = tmp_dir / f"{accession}.gb"
+
+        # Set the mock status code to 404 for request not found
+        mock_requests_failed.return_value.status_code = 404
+
+        # Raise an error 
+        with raises(DownloadError):
+            download_genbank(accession, output_file)
```

Comment:
Whilst sometimes adding empty lines can help read the code, I believe this code is simple enough to not need them:
```suggestion
        output_file = tmp_dir / f"{accession}.gb"
        # Set the mock status code to 404 for request not found
        mock_requests_failed.return_value.status_code = 404
        with raises(DownloadError):
            download_genbank(accession, output_file)
```